mktrayicon.c

```diff
@@ -320,79 +328,92 @@ gpointer watch_fifo(gpointer argv)
 #ifdef DEBUG
 				printf("Removing onmenu handler\n");
 #endif
+				free(param);
 				break;
 			}
 
-			// This block makes sure that the parameter after 'm' is well defined
-			int commas = 0;
-			int bars = 0;
-			if (len < 3){
-				break;
-			}
-			if (param[0] == ',' || param[0] == '|'){
-				printf("Action string cannot begin with ',' or '|'. Use '\\' to escape\n");
-				break;
-			}
-			if (param[len-2] == '|' && param[len-3] != '\\'){
-				printf("'|' can't be second of last. Use '\\' to escape\n");
-				break;
+			// This block makes sure that the parameter after 'm' is ready to be processed
+			
+			char* initial = param; 
+			//we save so that we can free later even if we increase the pointer
+			
+			while (param[0] == ',' || param[0] == '|'){
+				// ignore delimiter characters in first spot
+				param++;
+				len--;
 			}
-			if (param[len-1] == ',' && param[len-2] != '\\'){
-				printf("',' can't be last. Use '\\' to escape\n");
-				break;
+			while ((param[len-1] == ',' || param[len-1] == '|') && param[len-2] != '\\'){
+				// ignore delimiters at the end too
+				// it also helps dividing the string in entries -> bars+1
```

Comment:
Same here, I don't think we should treat the beginning and end of the string specially.